pulp_ansible/app/models.py

```diff
@@ -467,8 +490,8 @@ def finalize_new_version(self, new_version):
             base_version=new_version.base_version
         ).filter(pulp_type=CollectionVersion.get_pulp_type())
 
-        # Remove any deprecated, signature and namespace content associated with the removed
-        # collection versions
+        # Remove any deprecated, signature, mark and namespace content
+        # associated with the removed collection versions
         for version in removed_collection_versions:
             version = version.cast()
```

Comment:
Hrm. All we care for in the c_version is the pk. Right? I don't think we need to `cast` for that. And at that point we don't even need the loop anymore, right?
Would the following work?
```python
        signatures = new_version.get_content(
            content_qs=CollectionVersionSignature.objects.filter(signed_collection__in=removed_collection_versions)
        )
        new_version.remove_content(signatures)
```

Reply:
Also do we maybe need to call `remove_duplicates` first? It may remove more CVs.

Reply:
I prefer to handle this in another PR, my changes are not touching this logic, I am just following the same pattern for the marks, what if we open a new issue based on this comment and then fix in another pr?